navigation-animation/src/main/java/com/google/accompanist/navigation/animation/AnimatedNavHost.kt

```diff
@@ -174,43 +169,27 @@ public fun AnimatedNavHost(
             val targetDestination = targetState.destination as AnimatedComposeNavigator.Destination
 
             if (composeNavigator.isPop.value) {
-                targetDestination.popEnterTransition?.invoke(this)
-                    ?: popEnterTransitions[
-                        (
-                            targetDestination.hierarchy.first {
-                                popEnterTransitions.containsKey(it.route)
-                            }
-                            ).route
-                    ]?.invoke(this) as EnterTransition
+                targetDestination.hierarchy.mapNotNull { destination ->
+                    popEnterTransitions[destination.route]?.invoke(this)
+                }.firstOrNull() ?: popEnterTransition.invoke(this)
```

Comment:
Isn't this a 
```kotlin
 targetDestination.hierarchy.firstNotNullOfOrNull { destination ->
     popEnterTransitions[destination.route]?.invoke(this)
 } ?: popEnterTransition.invoke(this)
```
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/first-not-null-of-or-null.html

Reply:
replaced it. Thanks for the suggestion!